kaos/algorithm/view_cone.py

```diff
@@ -1,164 +1,163 @@
 """Implementation of Viewing cone algorithm"""
-from math import asin, atan, sqrt, sin, cos, pi
-import numpy as np
-from numpy import cross
-from numpy.linalg import norm
 
-from ..constants import SECONDS_PER_DAY, ANGULAR_VELOCITY_EARTH, THETA_NAUGHT
+from __future__ import division
+
+import mpmath as mp
+from numpy import cross, asarray
+
+from .coord_conversion import geod_to_geoc_lat, geod_to_eci_geoc_lon
+from ..utils import interval_utils
+from ..constants import ANGULAR_VELOCITY_EARTH, THETA_NAUGHT, EARTH_A_AXIS, EARTH_B_AXIS
 from ..tuples import TimeInterval
 from ..errors import ViewConeError
 
 
-# pylint: disable=invalid-name,too-many-locals
-def cart2sp(x, y, z):
-    """Converts data from cartesian coordinates into spherical.
+def reduce_poi(site_lat_lon, sat_pos_list, sat_vel_list, q_max, poi):
```

Comment:
Instead of taking, 2 lists and asserting that they have equal elements, just take in a list of 2-tuples